### PR TITLE
amplitude based stimuli in apply_multiple_stimuli

### PR DESCRIPTION
**Pull request by AurelienJaquier**
can choose to use `apply_multiple_stimuli` with amplitude based stimuli by passing raw amplitudes to `amplitudes` and setting `threshold_based` to `False`.

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
It looks like my additions to the tests increased the ci testing time. :')
Do you think my additional tests are worth the time @anilbey ?

File: tests/test_analysis/test_inject_sequence.py [hunk 1]
```diff
@@ -32,16 +32,19 @@ def mock_run_stimulus():
 def test_apply_multiple_step_stimuli(mock_run_stimulus):
     """Do not run the code in parallel, mock the return value via MockRecording."""
     amplitudes = [80, 100, 120, 140]
+    thres_perc = [0.08, 0.1, 0.12, 0.14]
```
Comment (anilbey) [inline tests/test_analysis/test_inject_sequence.py:35]: For test time maybe instead of 4, just 1 or 2 values can be kept here

File: bluecellulab/stimulus/factory.py [hunk 2]
```diff
@@ -299,106 +300,167 @@ def ramp(
...
+        if amplitude is not None:
+            return Step.amplitude_based(
+                self.dt,
+                pre_delay=pre_delay,
+                duration=duration,
+                post_delay=post_delay,
+                amplitude=amplitude,
+            )
+
+        raise TypeError("You have to give either threshold_current or amplitude")
```
Comment (anilbey) [inline bluecellulab/stimulus/factory.py:338]: I like to catch these edge cases in the tests. It may look trivial but there are sometimes issues due to those.

File: bluecellulab/stimulus/factory.py [hunk 3]
```diff
@@ -299,106 +300,167 @@ def ramp(
...
 
     def idrest(
         self,
-        threshold_current: float,
-        threshold_percentage: float = 200.0,
+        threshold_current: Optional[float] = None,
+        threshold_percentage: Optional[float] = 200.0,
+        amplitude: Optional[float] = None,
```
Comment (anilbey) [inline bluecellulab/stimulus/factory.py:344]: What if the user mistakenly provides all the 3 of `threshold_current`, `threshold_percentage` and `amplitude`?
Or what if both the `amplitude` and `threshold_current` are provided but the `threshold_percentage` missing?
Can those cases also be covered with logs/exceptions/warnings?

**Review by anilbey — APPROVED** [review]
Thanks looks good to me.

**Comment by AurelienJaquier** [comment]
Great! This should solve Issue #180
